ml-agents-envs/mlagents/envs/environment.py

```diff
@@ -373,7 +373,18 @@ def step(
         custom_action = {} if custom_action is None else custom_action
 
         # Check that environment is loaded, and episode is currently running.
-        if self._loaded and not self._global_done and self._global_done is not None:
+        if not self._loaded:
```

Comment:
It wasn't obvious that this block either raises or returns; I think this is a little cleaner.
Could also unindent the "main" block but trying to keep the diff smaller.